bft_types/src/lib.rs

```diff
@@ -97,53 +136,121 @@ impl Display for LocalisedInstruction {
 #[derive(Debug)]
 pub struct BfProgram {
     /// Name of the file containing the original program
-    name: PathBuf,
-    /// A vector of instructions. Not sure how else to desribe it
-    instructions: Vec<LocalisedInstruction>,
+    name: PathBuf,
+    /// A vector of instructions. Not sure how else to describe it
+    instructions: Vec<LocalisedInstruction>,
+    /// Hashmap to record the index of jumps in instructions, and the indexes of their counterparts
+    jump_map: HashMap<usize, usize>,
```

Comment:
This hashmap seems very expensive - given there will be potentially tens of millions of lookups happening in some of my test cases which I'd expect to run in a few tens or at most a few hundreds of milliseconds, something which is cheaper to index into would be a more appropriate .